scripts/ci-k8s-common.sh

```diff
@@ -0,0 +1,186 @@
+#!/bin/bash
+
+# AZURE_RESOURCE_GROUP will be set by the calling module
+set -o errexit
+set -o nounset
+set -o pipefail
+set -o errtrace
+
+LOCAL_DIR=${BASH_SOURCE[0]}
+SSH_OPTS="-o ServerAliveInterval=20 -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null"
+AZURE_IMG="${WIN_VM_IMG:-$AZURE_DEFAULT_IMG}"
+VM_NAME="winTestVM"
+VM_LOCATION="${VM_LOCATION:-westus2}"
+VM_SIZE="${VM_SIZE:-Standard_D2s_v3}"
+ARTIFACTS="${ARTIFACTS:-/var/log/artifacts}"
+
+function onError(){
+    az group list | jq ' .[] | .name' | grep ${AZURE_RESOURCE_GROUP}
+    if [ $? -ne 0 ]; then
+        az group delete -n ${AZURE_RESOURCE_GROUP} --yes --no-wait
+    fi
```

Comment:
It might be easier to just do:
```suggestion
   az group delete -n ${AZURE_RESOURCE_GROUP} --yes --no-wait || true
```
You could also just call `destroy_resource_group` here.

Reply:
This is reverting back to the original behavior, which will only delete the RG when there is no RG being found, so basically did nothing. I have tried to fix this but it looks it will cause issues where RG is deleted unexpectedly. As the old behavior has been there and "working" for a long time, I think I will not include this in this PR. We can address this in a separate PR if we want

Reply:
yes, let's fix this in a separate PR